scripts/eurostat/life_expectancy/generate_mcf.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pandas as pd
+import re
+
+def convert_range(s_input):
+    """Convert range values from the format in statvar names (e.g. 10YearsOrMore)
+     to the format as QuantityRange (e.g. [Years 10 -]) in Data Commons."""
+    if 'OrMore' in s_input:
+        match = re.match(r'(\d+)OrMore([a-zA-Z]+)', s_input)
+        num = match.group(1)
+        unit = match.group(2)
+        return '[{} {} -]'.format(unit, num)
+    elif 'Upto' in s_input:
+        match = re.match(r'Upto(\d+)([a-zA-Z]+)', s_input)
+        num = match.group(1)
+        unit = match.group(2)
+        return '[{} - {}]'.format(unit, num)
+    elif 'To' in s_input:
+        match = re.match(r'(\d+)To(\d+)([a-zA-Z]+)', s_input)
+        num1 = match.group(1)
+        num2 = match.group(2)
+        unit = match.group(3)
+        return '[{} {} {}]'.format(unit, num1, num2)
+    else:
+        match = re.match(r'(\d+)([a-zA-Z]+)', s_input)
+        num = match.group(1)
+        unit = match.group(2)
+        return '[{} {}]'.format(unit, num)
+  
+def generate_statvar(statvars, path):
+    """Generate the statvars from the list of statvar names."""
+    by_age_template = ('Node: dcid:{stat_var}\n'
+                       'typeOf: dcs:StatisticalVariable\n'
+                       'populationType: schema:Person\n'
+                       'age: {age}\n'
+                       'measuredProperty: dcs:lifeExpectancy\n'
+                       'statType: dcs:measuredValue\n\n')
+                       
+    by_age_gender_template = by_age_template[:-1] + 'gender: schema:{gender}\n\n'
+
+    with open(path, 'w') as f_out:
+        for stat_var in statvars:
+            keys = stat_var.split('_')
+            age = convert_range(keys[2])
```

Comment:
We should not really ask for the [2] value until we know the `keys` list has at least length three.  If the length is shorter, this will crash.

Reply:
fixed